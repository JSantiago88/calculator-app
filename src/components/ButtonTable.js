import React from 'react';
import PropTypes from 'prop-types';

import ButtonRow from './ButtonRow';
import { data as rows } from '../data';

const ButtonTable = ({ handleSelectButton }) => {

    return (
        <div className="button-table">
            {
                rows.map(({ row, values }) => (
                    <ButtonRow
                        key={row}
                        buttons={values}
                        handleSelectButton={handleSelectButton}
                    />
                ))
            }
        </div>
    );
};

ButtonTable.propTypes = {
    handleSelectButton: PropTypes.func.isRequired,
}

export default ButtonTable;