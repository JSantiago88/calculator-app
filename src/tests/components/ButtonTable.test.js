import React from 'react';
import { shallow } from 'enzyme';

import ButtonTable from '../../components/ButtonTable';

describe('Test in <ButtonTable />', () => {

   
    const handleSelectButton = () => { };

    const wrapper = shallow(<ButtonTable handleSelectButton={handleSelectButton} />);

    test('should ButtonTable the component correctly', () => {

        expect(wrapper).toMatchSnapshot();
    });


    test('should have a css class in the div', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');

        expect(className.includes('button-table')).toBe(true);
    });

});
