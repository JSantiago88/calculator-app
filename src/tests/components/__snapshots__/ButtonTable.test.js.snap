// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ButtonTable /> should ButtonTable the component correctly 1`] = `
<div
  className="button-table"
>
  <ButtonRow
    buttons={
      Array [
        Object {
          "styles": "btn",
          "value": "+",
        },
        Object {
          "styles": "btn",
          "value": "-",
        },
        Object {
          "styles": "btn",
          "value": "x",
        },
        Object {
          "styles": "btn",
          "value": "/",
        },
      ]
    }
    handleSelectButton={[Function]}
    key="1"
  />
  <ButtonRow
    buttons={
      Array [
        Object {
          "styles": "btn",
          "value": "7",
        },
        Object {
          "styles": "btn",
          "value": "8",
        },
        Object {
          "styles": "btn",
          "value": "9",
        },
        Object {
          "styles": "btn btn-blue",
          "value": "=",
        },
      ]
    }
    handleSelectButton={[Function]}
    key="2"
  />
  <ButtonRow
    buttons={
      Array [
        Object {
          "styles": "btn",
          "value": "4",
        },
        Object {
          "styles": "btn",
          "value": "5",
        },
        Object {
          "styles": "btn",
          "value": "6",
        },
      ]
    }
    handleSelectButton={[Function]}
    key="3"
  />
  <ButtonRow
    buttons={
      Array [
        Object {
          "styles": "btn",
          "value": "1",
        },
        Object {
          "styles": "btn",
          "value": "2",
        },
        Object {
          "styles": "btn",
          "value": "3",
        },
      ]
    }
    handleSelectButton={[Function]}
    key="4"
  />
  <ButtonRow
    buttons={
      Array [
        Object {
          "styles": "btn",
          "value": "0",
        },
        Object {
          "styles": "btn",
          "value": ",",
        },
        Object {
          "styles": "btn btn-red",
          "value": "AC",
        },
      ]
    }
    handleSelectButton={[Function]}
    key="5"
  />
</div>
`;
